### DISCUSSION TITLE
How important is to use userflows for Single Page Applications. How prevalent are userflows being used in market? Does google use it for its SPA

### DISCUSSION THREAD

**Discussion by SivasubramanianV** [GoogleChrome/lighthouse#15350] [Q&A] [answered]
How important is to use userflows for Single Page Applications. There are a lot of transitions in my application where there are url changes but no page load. How do i explain the importance of capturing timespan and snapshots data to the team for increasing the overall lighthouse score? Some discount that in a SPA all the content are loaded upfront and there is no necessity to check for timespan and snapshot but i see a lot of CLS and TBT at the final steps too. I feel timespan and snapshot provide great insight, any strong points to back it up?

**Comment by adamraine** [comment, marked as answer]
Post-navigation performance is still important even if there isn't an overall score for it. This is the entire reason we added user flows in the first place :)

CLS and TBT still causes negative user experiences after the page load, and timespan mode is the only way to measure INP. We are even planning on adding a new mode just for SPA transitions (AKA soft navigations) https://github.com/GoogleChrome/lighthouse/issues/14573.

It doesn't have to be related to SPA transitions easily. Snapshot mode allows you to check for accessibility violations on some UX reveled through user interactions.
